files/image_config/hostcfgd/hostcfgd

```diff
@@ -268,38 +268,41 @@ class HostConfigDaemon:
             if not key:
                 syslog.syslog(syslog.LOG_WARNING, "FEATURE key is missing")
                 return
-        status = status_data[key]['status']
-        if not status:
-            syslog.syslog(syslog.LOG_WARNING, "status is missing for {}".format(key))
-            return
-        if status == "enabled":
-            start_cmds=[]
-            start_cmds.append("sudo systemctl enable {}".format(key))
-            start_cmds.append("sudo systemctl start {}".format(key))
-            for cmd in start_cmds:
-                syslog.syslog(syslog.LOG_INFO, "Running cmd - {}".format(cmd))
-                try:
-                    subprocess.check_call(cmd, shell=True)
-                except subprocess.CalledProcessError as err:
-                    syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}"
-                              .format(err.cmd, err.returncode, err.output))
-                    return
-            syslog.syslog(syslog.LOG_INFO, "Feature '{}' is enabled and started".format(key))
-        elif status == "disabled":
-            stop_cmds=[]
-            stop_cmds.append("sudo systemctl stop {}".format(key))
-            stop_cmds.append("sudo systemctl disable {}".format(key))
-            for cmd in stop_cmds:
-                syslog.syslog(syslog.LOG_INFO, "Running cmd - {}".format(cmd))
-                try:
-                    subprocess.check_call(cmd, shell=True)
-                except subprocess.CalledProcessError as err:
-                    syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}"
-                              .format(err.cmd, err.returncode, err.output))
-                    return
-            syslog.syslog(syslog.LOG_INFO, "Feature '{}' is stopped and disabled".format(key))
-        else:
-            syslog.syslog(syslog.LOG_ERR, "Unexpected status value '{}' for '{}'".format(status, key))
+
+            status = status_data[key]['status']
+            if not status:
+                syslog.syslog(syslog.LOG_WARNING, "status is missing for {}".format(key))
+                return
+            if status == "enabled":
+                start_cmds=[]
+                start_cmds.append("sudo systemctl unmask {}.service".format(key))
+                start_cmds.append("sudo systemctl enable {}.service".format(key))
+                start_cmds.append("sudo systemctl start {}.service".format(key))
+                for cmd in start_cmds:
+                    syslog.syslog(syslog.LOG_INFO, "Running cmd - {}".format(cmd))
+                    try:
+                        subprocess.check_call(cmd, shell=True)
+                    except subprocess.CalledProcessError as err:
+                        syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}"
```

Comment:
I think since you intended, you may want to `continue `here instead of `return`? Also since mask/unmask is introduced and if it fails for some reason, you can still `continue `with other services. (Same for other `returns `in this block as well). What do you think?

Reply:
Yes, I think it makes more sense to continue rather than bail. Thanks, Prince!